### PR TITLE
Upgrade go crypto and adapt SEIPD

### PR DESCRIPTION
**Pull request by wussler**

### CODE REVIEW

**Comment by wussler** [comment]
This issue closes #210

**Review by meshenka — APPROVED** [review]

File: crypto/sessionkey.go [hunk 1]
```diff
@@ -239,7 +239,15 @@ func encryptStreamWithSessionKey(
 	signEntity *openpgp.Entity,
 	config *packet.Config,
 ) (encryptWriter, signWriter io.WriteCloser, err error) {
-	encryptWriter, err = packet.SerializeSymmetricallyEncrypted(dataPacketWriter, config.Cipher(), sk.Key, config)
+	encryptWriter, err = packet.SerializeSymmetricallyEncrypted(
+		dataPacketWriter,
+		config.Cipher(),
```
Comment (marinthiercelin) [inline crypto/sessionkey.go:244]: Why do we need the cipher both in here and in the ciphersuite ?

Comment (wussler) [reply crypto/sessionkey.go:244]: In this case they're taken from the config, but in a key you could have different preferences for AEAD and for the normal cipher. So we need a generic interface that allows to pass different settings here

See [the SEIPD v1 preferences](https://www.ietf.org/archive/id/draft-ietf-openpgp-crypto-refresh-07.html#name-preferred-symmetric-ciphers) vs [the SEIPD v2 preferences](https://www.ietf.org/archive/id/draft-ietf-openpgp-crypto-refresh-07.html#name-preferred-aead-ciphersuites)

Comment (marinthiercelin) [reply crypto/sessionkey.go:244]: But for a single call to SerializeSymetricallyEncrypted, you would use only one cipher right ?

Comment (wussler) [reply crypto/sessionkey.go:244]: Yes, correct, the picked cipher depends whether AEAD is enabled or not

**Review by marinthiercelin — APPROVED** [review]

**Pull request merged**